packages/astro/src/@types/astro.ts

```diff
@@ -822,6 +822,61 @@ export interface AstroUserConfig {
 		 * ```
 		 */
 		inlineStylesheets?: 'always' | 'auto' | 'never';
+
+		/**
+		 * @docs
+		 * @name build.serverlessEntryPrefix
+		 * @type {string}
+		 * @default `'entry'`
+		 * @description
+		 * Option used when `build.mode` is set to `"serverless"`. Astro will prepend
+		 * the emitted files using this option.
+		 *
+		 * ```js
+		 * {
+		 *   build: {
+		 *     serverlessEntryPrefix: 'main'
+		 *   }
+		 * }
+		 * ```
+		 */
+		serverlessEntryPrefix?: string;
```

Comment:
After talking offline we thought it might be premature to offer this option. Users who want to customize the filename probably want full customization. So we decided to omit this for now and then to add it later when/if people ask for it.